servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ProxyTunnel.java

```diff
@@ -86,7 +85,7 @@ HostAndPort startProxy() throws IOException {
             }
             return null;
         });
-        return HostAndPort.of((InetSocketAddress) serverSocket.getLocalSocketAddress());
+        return HostAndPort.of(getLoopbackAddress().getHostAddress(), serverSocket.getLocalPort());
```

Comment:
```suggestion
        InetSocketAddress serverSocketAddress = (InetSocketAddress) serverSocket.getLocalSocketAddress();
        return HostAndPort.of(serverSocketAddress.getAddress().getHostAddress(), serverSocketAddress.getPort());
```

We should still prefer using `serverSocket`, but query different IP to get the resolved IP.